superset/datasets/commands/bulk_delete.py

```diff
@@ -1,60 +0,0 @@
-# Licensed to the Apache Software Foundation (ASF) under one
-# or more contributor license agreements.  See the NOTICE file
-# distributed with this work for additional information
-# regarding copyright ownership.  The ASF licenses this file
-# to you under the Apache License, Version 2.0 (the
-# "License"); you may not use this file except in compliance
-# with the License.  You may obtain a copy of the License at
-#
-#   http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing,
-# software distributed under the License is distributed on an
-# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-# KIND, either express or implied.  See the License for the
-# specific language governing permissions and limitations
-# under the License.
-import logging
-from typing import Optional
-
-from superset import security_manager
-from superset.commands.base import BaseCommand
-from superset.commands.exceptions import DeleteFailedError
-from superset.connectors.sqla.models import SqlaTable
-from superset.daos.dataset import DatasetDAO
-from superset.datasets.commands.exceptions import (
-    DatasetBulkDeleteFailedError,
-    DatasetForbiddenError,
-    DatasetNotFoundError,
-)
-from superset.exceptions import SupersetSecurityException
-
-logger = logging.getLogger(__name__)
-
-
-class BulkDeleteDatasetCommand(BaseCommand):
-    def __init__(self, model_ids: list[int]):
-        self._model_ids = model_ids
-        self._models: Optional[list[SqlaTable]] = None
-
-    def run(self) -> None:
-        self.validate()
-        assert self._models
-
-        try:
-            DatasetDAO.delete(self._models)
-        except DeleteFailedError as ex:
```

Comment:
This was another positive byproduct of the change. Here we're catching a `DeleteFailedError` whereas in the non-bulk we were catching a `DAODeleteFailedError` error. The later is more accurate as it's a DAO rather than command failure.